### DISCUSSION TITLE
configuring-urls-with-kestrel-iis-and-iis-express-with-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#57] [Announcements]
# configuring-urls-with-kestrel-iis-and-iis-express-with-asp-net-core/

In this post I describe how to configure the URLs your application binds to when using Kestrel or the WebListener HTTP servers

https://andrewlock.net/configuring-urls-with-kestrel-iis-and-iis-express-with-asp-net-core/?utm_source=DNK-156061&utm_medium=DNK-156061&utm_content=DNK-156061&utm_campaign=DNK-156061

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 27 2016, 07:38, in reply to [Edgar Santiago](https://disqus.com/by/disqus_C1FRIyocGi/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/57#discussioncomment-5795154)</em>

---

<p>It depends to be honest. I tend to just run straight from the command line as I'm often on a Mac, so the command line approach is sometimes nicest as it keeps the config out of the code but is easy to specify. Other times I will just hard code the urls using UseUrls to make it explicit.</p><p>When it comes to actually running in production you should be behind IIS as a reverse proxy - see here Rick Strahl's blog for a post on doing that: <a href="https://weblog.west-wind.com/posts/2016/Jun/06/Publishing-and-Running-ASPNET-Core-Applications-with-IIS" rel="nofollow noopener" title="https://weblog.west-wind.com/posts/2016/Jun/06/Publishing-and-Running-ASPNET-Core-Applications-with-IIS">https://weblog.west-wind.com/posts/2016/Jun/06/Publishing-and-Running-ASPNET-Core-Applications-with-IIS</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at October 01 2016, 10:42, in reply to [Frank Ouimette](https://disqus.com/by/frank_ouimette/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/57#discussioncomment-5795160)</em>

---

<p>Something like this is pretty simple, obviously you can add more layers to it, pull it out into a Startup Class etc, but will prob get you what you want.</p><p></p><pre><code><br>using System.IO;<br>using Microsoft.AspNetCore.Builder;<br>using Microsoft.AspNetCore.Hosting;<br>using Microsoft.AspNetCore.Http;<br>using Microsoft.Extensions.DependencyInjection;<br>using Microsoft.Extensions.Logging;<br><br>namespace ConsoleWriter<br>{<br>    public class Program<br>    {<br>        public static void Main(string[] args)<br>        {<br>            new WebHostBuilder()<br>                .UseKestrel()<br>                .UseContentRoot(Directory.GetCurrentDirectory())<br>                .Configure(app =&gt;<br>                {<br>                    app.ApplicationServices.GetService&lt;iloggerfactory&gt;().AddConsole();<br>                    app.Run(async ctx =&gt;<br>                    {<br>                        ctx.RequestServices<br>                            .GetRequiredService&lt;ilogger&lt;program&gt;&gt;()<br>                            .LogInformation(GetInfoToLog(ctx));<br>                        await ctx.Response.WriteAsync("Request received");<br>                    });<br>                })<br>                .Build()<br>                .Run();<br>        }<br><br>        private static string GetInfoToLog(HttpContext context)<br>        {<br>            return context.Request.Path;<br>        }<br>    }<br>}<br><br></code></pre>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 10 2016, 09:49, in reply to [Andrej Matijevic](https://disqus.com/by/andrejmatijevic/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/57#discussioncomment-5795162)</em>

---

<p>Good point, this is actually true for all of the cases in fact, if you have changed the environment variables, open a new command prompt/terminal window/powershell window to reload the variables. Thanks</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 21 2017, 07:14, in reply to [Shimmy](https://disqus.com/by/weitzhandler/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/57#discussioncomment-5795172)</em>

---

<p>AFAIK, there's no way to add/remove URLs without having to restart the app.</p>

[a comment by an automated account is not shown]
